timeseries/src/autogluon/timeseries/learner.py

```diff
@@ -198,14 +202,58 @@ def score(
 
     def evaluate(
         self,
-        data: Union[TimeSeriesDataFrame, pd.DataFrame, str],
+        data: TimeSeriesDataFrame,
         model: Optional[str] = None,
         metrics: Optional[Union[str, TimeSeriesScorer, List[Union[str, TimeSeriesScorer]]]] = None,
         use_cache: bool = True,
     ) -> Dict[str, float]:
         data = self.feature_generator.transform(data)
         return self.load_trainer().evaluate(data=data, model=model, metrics=metrics, use_cache=use_cache)
 
+    def get_feature_importance(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Optional[str] = None,
+        metric: Optional[Union[str, TimeSeriesScorer]] = None,
+        features: Optional[List[str]] = None,
+        time_limit: Optional[float] = None,
+        method: Literal["naive", "permutation"] = "permutation",
+        subsample_size: int = 50,
+        num_iterations: int = 1,
+    ) -> pd.DataFrame:
+        trainer = self.load_trainer()
+        data = self.feature_generator.transform(data)
+
+        assert method in ["naive", "permutation"], f"Invalid feature importance method {method}."
+
+        if features is not None:
+            for fn in features:
+                if fn not in self.feature_generator.covariate_metadata.all_features:
+                    raise ValueError(f"Feature {fn} not found in covariate metadata.")
+
+        feature_importance_transform = (
+            {
+                "permutation": PermutationFeatureImportanceTransform,
+                "naive": ConstantReplacementFeatureImportanceTransform,
+            }
+            .get(method)
+            .from_feature_generator(self.feature_generator)
```

Comment:
Maybe we can keep all logic related to feature importance inside `trainer.get_feature_importance`? We should have all the necessary information available there (e.g., `covariate_metadata` is available inside the trainer as `self.covariate_metadata`). This would reduce the mental load on developers if we decide to extend this functionality in the future.